src/SFA.DAS.LevyTransferMatching/Application/Queries/Pledges/GetAmount/GetAmountQueryHandler.cs

```diff
@@ -1,26 +1,45 @@
-﻿using System.Threading;
+﻿using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 using MediatR;
+using SFA.DAS.LevyTransferMatching.Application.Queries.Pledges.GetPledges;
 using SFA.DAS.LevyTransferMatching.Interfaces;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests;
 
 namespace SFA.DAS.LevyTransferMatching.Application.Queries.Pledges.GetAmount
 {
     public class GetAmountQueryHandler : IRequestHandler<GetAmountQuery, GetAmountQueryResult>
     {
         private readonly IAccountsService _accountsService;
+        private readonly ILevyTransferMatchingService _levyTransferMatchingService;
 
-        public GetAmountQueryHandler(IAccountsService accountsService)
+        public GetAmountQueryHandler(IAccountsService accountsService, ILevyTransferMatchingService levyTransferMatchingService)
         {
             _accountsService = accountsService;
+            _levyTransferMatchingService = levyTransferMatchingService;
         }
 
         public async Task<GetAmountQueryResult> Handle(GetAmountQuery request, CancellationToken cancellationToken)
         {
-            var account = await _accountsService.GetAccount(request.EncodedAccountId);
+            // Get the current amount committed to existing pledges first
+            var response = await _levyTransferMatchingService.GetPledges(new GetPledgesRequest(request.AccountId));
+
+            GetPledgesQueryResult getPledgesQueryResult = new GetPledgesQueryResult
+            {
+                Pledges = response?.Pledges.Select(x => new GetPledgesQueryResult.Pledge
+                {
+                    Amount = x.Amount
+                })
+            };
+
+            int currentTransferAmount = getPledgesQueryResult?.Pledges?.Count() > 0 ? getPledgesQueryResult.Pledges.Sum(p => p.Amount) : 0;
```

Comment:
I think this could be rewritten as `getPledgesQueryResult.Pledges.Sum(p => p.Amount);` _if_ you change line 29 so that it does not tolerate `null` responses. Instead, we could check for a null response and return null from this method in this event - then, in the controller, we can check if the query result is null and return a NotFound() if so. (Some of this suggestion does not directly relate to your code change so apologies, but this would be a good time to pick it up).

Reply:
Done as suggested.